data_source_ssh_tunnel.go

```diff
@@ -98,19 +157,22 @@ func dataSourceSSHTunnelRead(d *schema.ResourceData, meta interface{}) error {
 	log.Printf("[DEBUG] tunnelEstablished: %v", tunnelEstablished)
 
 	if tunnelEstablished == false {
-		d.Set("tunnel_established", true)
 
 		sshConf := &ssh.ClientConfig{
 			User:            user,
 			HostKeyCallback: ssh.InsecureIgnoreHostKey(),
 			Auth:            []ssh.AuthMethod{},
 		}
 
-		pubKeyAuth, err := readPrivateKey(privateKey)
-		if err != nil {
-			panic(err)
+		if sshAgent {
+			sshConf.Auth = append(sshConf.Auth, agentAuth())
```

Comment:
I haven't tried this yet, but I assume that we can add multiple auth methods? In that case, along with my comment above regarding defaulting `ssh_agent` to true, we could change this logic to add the ssh key first (if provided), and then secondly add the ssh agent unless `ssh_agent` has been set to false deliberately. What do you think?

Reply:
The description says that we can pass as many methods as needed but I'm unsure if it will proceed when the first one fails:
```
	// Auth contains possible authentication methods to use with the
	// server. Only the first instance of a particular RFC 4252 method will
	// be used during authentication.
	Auth []AuthMethod

Reply:
Double checked, it works as expected.